Implement autostart for all profilers

Geometror opened the pull request:
Salvages #70766. Co-authored by @stmSi
Closes https://github.com/godotengine/godot-proposals/issues/6017

Additional changes on top of the original PR:
- Changed `Button` to `CheckBox` for improved readability and consistency (also removed the "Autoplay" icon)
- Minor refactoring: Rename `set_pressed(...)` to `set_profiling(...)` since the original name is very odd (not perfect, but something like `start_profiling()` and `stop_profiling()` would require further refactoring which is needed, but not part of this PR)

<img width="252" alt="{A1188438-0365-4AD7-9EBD-5758AB82293D}" src="https://github.com/user-attachments/assets/2a337441-ceee-4b1f-849c-c8327e2938da">

akien-mga commented on modules/multiplayer/editor/editor_network_profiler.cpp:
```diff
@@ -42,6 +43,7 @@ void EditorNetworkProfiler::_bind_methods() {
 
 void EditorNetworkProfiler::_notification(int p_what) {
 	switch (p_what) {
+		case NOTIFICATION_ENTER_TREE:
```
This shouldn't be necessary, `THEME_CHANGED` always happens together with `ENTER_TREE`.

arkology commented on editor/debugger/editor_profiler.cpp:
```diff
@@ -633,6 +639,13 @@ EditorProfiler::EditorProfiler() {
+	autostart_checkbox->set_toggle_mode(true);
+	autostart_checkbox->set_text(TTR("Autostart"));
+	autostart_checkbox->set_pressed(EditorSettings::get_singleton()->get_project_metadata("debug_options", "autostart_profiler", false));
+	autostart_checkbox->connect("pressed", callable_mp(this, &EditorProfiler::_autostart_pressed));
```
I know _nothing_ about godot codebase, but looks like `"pressed"` should be replaced with `SceneStringName(pressed)`

Geometror commented:
Small update: 
- Addressed the review comments
- Removed the automatic debugger tab change(inside the debugger panel) after start since that makes workflow assumptions which can be annoying for many users; the now removed logic also switched to the rightmost tab if autostart is enabled for >=2 profilers.

KoBeWi commented:
Should the autostart be remembered between sessions? Currently it is, but if someone forgets they enabled it, they might run into accidental performance problems.

KoBeWi commented on editor/debugger/editor_visual_profiler.cpp:
```diff
@@ -747,6 +752,13 @@ EditorVisualProfiler::EditorVisualProfiler() {
+	autostart_checkbox->set_toggle_mode(true);
+	autostart_checkbox->set_text(TTR("Autostart"));
+	autostart_checkbox->set_pressed(EditorSettings::get_singleton()->get_project_metadata("debug_options", "autostart_visual_profiler", false));
+	autostart_checkbox->connect(SceneStringName(pressed), callable_mp(this, &EditorVisualProfiler::_autostart_pressed));
```
You could connect to `toggled` signal, which would also allow to make `autostart_checbkox` local.

KoBeWi commented on editor/debugger/editor_visual_profiler.cpp:
```diff
@@ -747,6 +752,13 @@ EditorVisualProfiler::EditorVisualProfiler() {
 	hb->add_child(clear_button);
 
+	autostart_checkbox = memnew(CheckBox);
+	autostart_checkbox->set_toggle_mode(true);
```
CheckBox has this by default.

Geometror committed:
Implement Autostart Feature for Profiler / Visual Profiler / Network Profiler

Co-authored-by: stmSi <stm1998sithumyo@gmail.com>

KoBeWi reviewed (approved):
Looks alright, but my concern from https://github.com/godotengine/godot/pull/96759#issuecomment-2341161452 still stands.

akien-mga commented:
> Should the autostart be remembered between sessions? Currently it is, but if someone forgets they enabled it, they might run into accidental performance problems.

Yeah I share that concern. For now I'd err on the side of _not_ remembering it between sessions, and we can re-assess if users request it.

Geometror commented:
I think the profiler autostart setting should be preserved between sessions. Consistent and complete state preservation of the editor aligns with what users typically expect. Silently resetting preferences can be frustrating, even if they're minor—these small annoyances can add up, especially for those who frequently need to restart the editor (which is not that uncommon).

While it's true that users might forget that autostart is enabled for profilers, potentially causing unexpected performance issues, this risk is present even within a single session. To address this, I suggest implementing a warning or modified icons for the "Run project", "Run current scene" and "Run specific scene" actions. This approach would provide clear visual cues and cover all scenarios.

arkology commented:
> I think the profiler autostart setting should be preserved between sessions.

Exactly.

> To address this, I suggest implementing a warning or modified icons for the "Run project", "Run current scene" and "Run specific scene" actions. This approach would provide clear visual cues and cover all scenarios.

Another way (I think this will be good idea) - highlight tabs with running profilers (not only autostarted but any running profilers).
If debugger tab is closed - highlight debugger tab. 
If debugger tab is opened - highlight running profilers tabs and don't highlight debugger tab. Also, don't highlight selected profiler tabs. 
How exaclty highlight is open question. Different font, different font color or icon addition.

akien-mga commented:
Thanks!

Geometror commented:
I opened https://github.com/godotengine/godot/pull/97492 which addresses the concerns about preserving this setting between sessions.
